modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/util/v1_0/SkuUtil.java

```diff
@@ -89,18 +95,23 @@ public static CPInstance addOrUpdateCPInstance(
 				replacementCPInstanceUuid =
 					discontinuedCPInstance.getCPInstanceUuid();
 			}
-		}
 
-		Calendar discontinuedCalendar = CalendarFactoryUtil.getCalendar(
-			serviceContext.getTimeZone());
+			if (sku.getDiscontinuedDate() != null) {
```

Comment:
```suggestion
                 int discontinuedDateMonth = 0;
                 int discontinuedDateDay = 0;
		int discontinuedDateYear = 0;
		
			if (sku.getDiscontinuedDate() != null) {
```

Reply:
@andrea-ale-sbarra but then they wouldn't be initialized if the product isn't set as discontinued, so when reaching the `addOrUpdateCPInstance` line at the end of the method, it would throw an exception as these variables are passed as parameters.